src/Razor/src/Microsoft.VisualStudio.LanguageServer.ContainedLanguage/LSPConstants.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.LanguageServer.ContainedLanguage
+{
+    internal class LSPConstants
+    {
+        /// <summary>
+        /// Should be used a property name on a virtual document buffer in order for the VS Platoform LanguageServerClient
+        /// infrastructure recognize it as a contained language buffer and init corresponding language server
+        /// </summary>
+        public const string ContainedLanguageMarker = "ContainedLanguageMarker";
```

Comment:
I actually don't think this is needed anymore by the LSP platform

Reply:
I'll double-check

---
In reply to: [451973398](https://github.com/dotnet/aspnetcore-tooling/pull/2201#discussion_r451973398) [](ancestors = 451973398)

Reply:
Yeah, you are right, it doesn't seem they are using it anymore since they refactored that code. Also it seems that we should change content type to inert *before* we dispose text document. See [https://devdiv.visualstudio.com/DevDiv/_git/VSLanguageServerClient?path=%2Fsrc%2Fproduct%2FRemoteLanguage%2FImpl%2FLanguageServerBufferManager.cs&version=GBdevelop&line=62&lineEnd=62&lineStartColumn=21&lineEndColumn=39&lineStyle=plain](https://devdiv.visualstudio.com/DevDiv/_git/VSLanguageServerClient?path=%2Fsrc%2Fproduct%2FRemoteLanguage%2FImpl%2FLanguageServerBufferManager.cs&version=GBdevelop&line=62&lineEnd=62&lineStartColumn=21&lineEndColumn=39&lineStyle=plain). It sucks for us to have to be aware of this implementation detail, but I'm not sure how else to go about it. 

---
In reply to: [452396418](https://github.com/dotnet/aspnetcore-tooling/pull/2201#discussion_r452396418) [](ancestors = 452396418,451973398)

Reply:
Do ITextView's take a dependency on that impl too when they dispose buffers and friends?

Reply:
Not explicitly, but they do it in that order. It's actually in TextDocData.Close (called by text view on close)

[https://devdiv.visualstudio.com/DevDiv/_git/VS-Platform?path=%2Fsrc%2FEditor%2FVisualStudio%2FImpl%2FBufferAdapter%2FIVsPersistDocData.cs&version=GBmaster&line=63&lineEnd=63&lineStartColumn=17&lineEndColumn=129&lineStyle=plain](https://devdiv.visualstudio.com/DevDiv/_git/VS-Platform?path=%2Fsrc%2FEditor%2FVisualStudio%2FImpl%2FBufferAdapter%2FIVsPersistDocData.cs&version=GBmaster&line=63&lineEnd=63&lineStartColumn=17&lineEndColumn=129&lineStyle=plain)

---
In reply to: [452411612](https://github.com/dotnet/aspnetcore-tooling/pull/2201#discussion_r452411612) [](ancestors = 452411612)